### ISSUE TITLE
Analytics: Track first user STX deposit

### ISSUE THREAD

**Issue by beguene** [blockstack/blockstack-app#2232]
We should track when the user receive STX for the first time : their balance goes from 0 STX to a positive number.
This event shall be triggered only once and not again if the balance goes back to 0 and again to a positive number.
This will help track the deposit rate of new users.

**Comment by beguene** [comment]
related https://github.com/hirosystems/stacks-wallet-web/issues/2034

**Comment by beguene** [comment]
# Test Cases

## Test 1:
Use 2 wallets: wallet 1 is a new wallet, wallet 2 is an existing wallet with some STX.

1. Create a new wallet 1
2. Send some STX from wallet 2 to wallet 1
Expected: The event **first_deposit** shall be seen in segment (with amount in micro stacks and type: 'stx')

## Test 2:
1. Using the same wallet 1, resend **ALL** STX to wallet 2 (so the balance of wallet 1 is back to 0)
Expected: no event shall be triggered

## Test 3:

1. Send again some STX from wallet 2 to wallet 1

Expected: no event shall be triggered.

**Comment by Eshwari007** [comment]
@beguene All the above test passed,however curious on the amount value when the first stx deposit on a new account:
I only transferred 1 STX ,hence the question
analytics.track('deposit_first_stx', {
  amount: '**20736470**',
  network: 'mainnet',
  route: '/',
  type: 'stx',
  usingDefaultHiroApi: true,
  version: '3.3.0.175'
});

cc @markmhx

**Comment by beguene** [comment]
I am not getting this issue after some first tests, but I think it can happen if the user receives multiple transactions and later opens the wallet (which will trigger the analytics with the current balance as amount). We can improve on that by looking into the tx list, but it's also not full-proof.
Since we only want to know whether the user got some initial STX or not (boolean) and since this PR has been well-tested, can we release as is, create another issue and fix this amount issue later if needed ? If we merge this now we would finally be able track our OKR.
@markmhx Wdyt ?

**Comment by markmhendrickson** [comment]
Yep let's release if the `amount` refinements will take time. The most important thing is triggering this event in the first place.

**Comment by beguene** [comment]
@Eshwari007  I removed the amount field. This is ready for QA.

**Comment by Eshwari007** [comment]
Thank you @beguene
